### ISSUE TITLE
catchsql: except* functions break stream chunking

### ISSUE THREAD

**Issue by nigredo-tori** [tpolecat/doobie#855]
Some functions in `doobie.util.catchsql` can be reasonably implemented using `.handleErrorWith`/`.recoverWith`, but [use `.attempt` instead](https://github.com/tpolecat/doobie/blob/dd5d681fc31b59fa3352ce2db0c890eb6a4b5e99/modules/core/src/main/scala/doobie/util/catchsql.scala#L33-L34). This makes them unsuitable for use with `fs2.io.Stream`, since `.attempt` breaks chunking:

```scala
scala> def getChunks[A](ma: Stream[SyncIO, A]) = ma.chunks.compile.toVector.unsafeRunSync
getChunks: [A](ma: fs2.Stream[cats.effect.SyncIO,A])Vector[fs2.Chunk[A]]

scala> val ma = Stream.emits(1 to 10).covary[SyncIO]
ma: fs2.Stream[cats.effect.SyncIO,Int] = Stream(..)

scala> getChunks(ma)
res19: Vector[fs2.Chunk[Int]] = Vector(Chunk(1, 2, 3, 4, 5, 6, 7, 8, 9, 10))

scala> getChunks(exceptSql(ma)(_ => Stream.empty))
res20: Vector[fs2.Chunk[Int]] = Vector(Chunk(1), Chunk(2), Chunk(3), Chunk(4), Chunk(5), Chunk(6), Chunk(7), Chunk(8), Chunk(9), Chunk(10))

scala> getChunks(ma.handleErrorWith(_ => Stream.empty))
res21: Vector[fs2.Chunk[Int]] = Vector(Chunk(1, 2, 3, 4, 5, 6, 7, 8, 9, 10))
```

This hurts a lot when using the blob streaming in PostgreSQL module.

I think the implementations of these functions should be changed to use more direct/performant approach.

**Comment by nigredo-tori** [comment]
Also, `doobie.syntax.MonadErrorOps.guarantee` can be easily mixed up with `Bracket.guarantee` in cats-effect, leading to all sorts of weird behavior (for example, try using it for a `Stream` with more than one element). It should be renamed or removed.

**Comment by tpolecat** [comment]
Yeah, thanks. Those are left over from the dark ages and need to be fixed for 0.7
